packages/cursorless-engine/src/core/mergeSnippets.test.ts

```diff
@@ -0,0 +1,390 @@
+            },
+          },
+        ],
+        description: "user snippet",
+      },
+    },
+  },
+
+  {
+    name: "should prefer more specific snippets, even if they are from a lower priority origin",
+    coreSnippets: {
+      aaa: {
+        definitions: [
+          {
+            body: ["core aaa"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+        ],
+        description: "core snippet",
+      },
+    },
+    userSnippets: [
+      {
+        aaa: {
+          definitions: [
+            {
+              body: ["user aaa"],
+            },
+          ],
+          description: "user snippet",
+        },
+      },
+    ],
+    expected: {
+      aaa: {
+        definitions: [
+          {
+            body: ["core aaa"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+          {
+            body: ["user aaa"],
+          },
+        ],
+        description: "user snippet",
+      },
+    },
+  },
+
+  {
+    name: "should prefer snippets with more specific languages",
+    coreSnippets: {
+      aaa: {
+        definitions: [
+          {
+            body: ["has no langIds"],
+          },
+          {
+            body: ["has two langIds"],
+            scope: {
+              langIds: ["typescript", "javascript"],
+            },
+          },
+          {
+            body: ["has one langId"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+        ],
+      },
+    },
+    expected: {
+      aaa: {
+        definitions: [
+          {
+            body: ["has one langId"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+          {
+            body: ["has two langIds"],
+            scope: {
+              langIds: ["typescript", "javascript"],
+            },
+          },
+          {
+            body: ["has no langIds"],
+          },
+        ],
+      },
+    },
+  },
+
+  {
+    name: "should prefer snippets with more specific languages, regardless of scopeType",
+    coreSnippets: {
+      aaa: {
+        definitions: [
+          {
+            body: ["has scope but no langIds"],
+            scope: {
+              scopeTypes: ["anonymousFunction"],
+            },
+          },
+          {
+            body: ["has langIds"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+        ],
+      },
+    },
+    expected: {
+      aaa: {
+        definitions: [
+          {
+            body: ["has langIds"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+          {
+            body: ["has scope but no langIds"],
+            scope: {
+              scopeTypes: ["anonymousFunction"],
+            },
+          },
+        ],
+      },
+    },
+  },
+
+  {
+    name: "should prefer snippets with more specific scope types",
```

Comment:
doesn't really matter anymore either, because we prefer snippets that matched a scope down the line anyway